changelog.d/14674.doc

```diff
@@ -0,0 +1 @@
+Fix `target_memory_usage` being used in the description for the actual `cache_autotune` sub-option `target_cache_memory_usage`.
```

Comment:
I think something like "Fix error in the configuration manual entry for `cache_autotune`" might be a little clearer?